python_modules/dagster/dagster/_core/definitions/asset_check_spec.py

```diff
@@ -81,12 +89,33 @@ def __new__(
         *,
         asset: Union[CoercibleToAssetKey, "AssetsDefinition", "SourceAsset"],
         description: Optional[str] = None,
+        additional_deps: Optional[Iterable["CoercibleToAssetDep"]] = None,
     ):
+        from dagster._core.definitions.asset_dep import AssetDep
+
+        asset_key = AssetKey.from_coercible_or_definition(asset)
+
+        additional_dep_set = {}
+        if additional_deps:
+            for dep in additional_deps:
+                asset_dep = AssetDep.from_coercible(dep)
+
+                # we cannot do deduplication via a set because MultiPartitionMappings have an internal
```

Comment:
I think probably worth refactoring to consolidate with the other place where we have this logic and explanation